app/scripts/components/toolbar/toolbar.components.jsx

```diff
@@ -63,7 +64,13 @@ class ViewButtons extends React.Component {
 	}
 
 	toggleView(name) {
-		const newViewMode = _.xor(this.state.mode, [name]);
+		// if we are toggling glyph mode, we want glyph list to be hidden
+		const modes = (
+			name === 'glyph'
+				? _.without(this.state.mode, 'list')
+				: this.state.mode
+		);
+		const newViewMode = _.intersection(_.xor(modes, [name]), this.availableMode);
```

Comment:
If available mode has all the possible mode it is useless to do intersection. Remove this.available and the intersection.